pkg/deploy/errors/error.go

```diff
@@ -74,8 +74,16 @@ type EnvironmentDeploymentErrors map[string][]error
 
 func (e EnvironmentDeploymentErrors) Error() string {
 	b := strings.Builder{}
+	b.WriteString(fmt.Sprintf("Errors encountered for %d environment(s):", len(e)))
 	for env, errs := range e {
-		b.WriteString(fmt.Sprintf("%s deployment errors: %v", env, errs))
+		if len(errs) == 1 {
```

Comment:
nit: I think we can just skip the `len()==1`case. Might make some errors a bit nicer, IMO not worth the condition here

Reply:
I don't think the condition makes this method overly complex does it?
As improving how errors are presented was part of what we set out to do here, I think presenting the error in a nicer is well worth it here. 

Note that this is generally the final error that is returned all the way to cobra and printed after execution.
I did only add the if in later, as it looked weird.